gcp_delete_old_images_and_vms.py

```diff
@@ -0,0 +1,102 @@
+                               'object value(family)', '--no-standard-images']
+    res = subprocess.run(families_cmd, capture_output=True,
+                         check=True, text=True)
+    families = set(res.stdout.split())
+
+    # find the newest image for each family
+    newest_family_members = set()
+    for family in families:
+        newest_cmd = base_cmd + ['describe-from-family', '--format',
+                                 'object value(name)', family]
+        res = subprocess.run(newest_cmd, capture_output=True,
+                             check=True, text=True)
+        newest_family_members.add(res.stdout.strip())
+
+    # get all old images, including the newest image in a
+    # family (will be skipped below)
+    old_images_cmd = base_cmd + ['list', '--format', 'object value(name)',
+                                 '--no-standard-images', '--filter',
+                                 'creationTimestamp < -P2W']
+    res = subprocess.run(old_images_cmd, capture_output=True,
+                         check=True, text=True)
+    old_images = res.stdout.split()
+
+    # filter to-be-deleted images by the newest image in a family
+    delete_images = []
+    for old_image in old_images:
+        if old_image in newest_family_members:
+            print(f"not deleting {old_image}, it's the newest family member")
+        else:
+            delete_images.append(old_image)
+
+    if len(delete_images) == 0:
+        print("no images to delete")
+        sys.exit(0)
+
+    print("deleting images: ", ', '.join(delete_images))
+
+    # finally delete old images
+    delete_cmd = base_cmd + ['delete', '--quiet'] + delete_images
+    subprocess.run(delete_cmd, check=True)
+
+
+# Delete VMs older than 23 hour
+def delete_vms():
+    base_cmd = ['gcloud', 'compute', '--project',
+                os.environ['GCP_PROJECT'], 'instances']
+
+    # get VMs older than 23 hours with their names and zones
+    # 23 hour is used instead of 1 day because cron job runs daily,
+    # so there could be race conditions
+    vms_cmd = base_cmd + ['list', '--format', 'object value(name, zone)',
+                          '--filter', 'creationTimestamp < -P23H']
+    res = subprocess.run(vms_cmd, capture_output=True,
+                         check=True, text=True)
+
+    delete_vms = []
+    for line in res.stdout.strip().split('\n'):
+        # check if line is not empty
+        if line:
+            linesplit = line.split()
+            delete_vms.append({'name': linesplit[0],
+                               'zone': linesplit[1]})
+
+    if delete_vms:
+        print("VMs to be deleted: ", ', '.join(
+            vm['name']for vm in delete_vms))
+
+    # delete old VMs
+    for vm in delete_vms:
+        print('Deleting VM =', vm['name'])
+        delete_cmd = base_cmd + ['delete', vm['name'],
+                                 '--zone', vm['zone'], '--quiet']
+        subprocess.run(delete_cmd, check=True)
```

Comment:
I think we also need to delete old disks for those VMs?

Reply:
This command actually deletes VMs and their disks.

Reply:
I think that's only true for disks explicitly owned by VMs - but the way packer creates disk they are not set up that way (because packer needs to access it after shutting the VM down).

Reply:
Added a explicit way to deleting remaining disks older than 23 hours